src/Protocol/ActivityPub/Transmitter.php

```diff
@@ -0,0 +1,843 @@
+			$context_uri = $conversation['conversation-uri'];
+		} else {
+			$context_uri = str_replace('/objects/', '/context/', $item['parent-uri']);
+		}
+		return $context_uri;
+	}
+
+	/**
+	 * @brief Creates a note/article object array
+	 *
+	 * @param array $item
+	 *
+	 * @return object array
+	 */
+	private static function createNote($item)
+	{
+		if (!empty($item['title'])) {
+			$type = 'Article';
+		} else {
+			$type = 'Note';
+		}
+
+		if ($item['deleted']) {
+			$type = 'Tombstone';
+		}
+
+		$data = [];
+		$data['id'] = $item['uri'];
+		$data['type'] = $type;
+
+		if ($item['deleted']) {
+			return $data;
+		}
+
+		$data['summary'] = null; // Ignore by now
+
+		if ($item['uri'] != $item['thr-parent']) {
+			$data['inReplyTo'] = $item['thr-parent'];
+		} else {
+			$data['inReplyTo'] = null;
+		}
+
+		$data['diaspora:guid'] = $item['guid'];
+		$data['published'] = DateTimeFormat::utc($item["created"]."+00:00", DateTimeFormat::ATOM);
+
+		if ($item["created"] != $item["edited"]) {
+			$data['updated'] = DateTimeFormat::utc($item["edited"]."+00:00", DateTimeFormat::ATOM);
+		}
+
+		$data['url'] = $item['plink'];
+		$data['attributedTo'] = $item['author-link'];
+		$data['actor'] = $item['author-link'];
+		$data['sensitive'] = false; // - Query NSFW
+		$data['context'] = self::fetchContextURLForItem($item);
+
+		if (!empty($item['title'])) {
+			$data['name'] = BBCode::convert($item['title'], false, 7);
+		}
+
+		$data['content'] = BBCode::convert($item['body'], false, 7);
+		$data['source'] = ['content' => $item['body'], 'mediaType' => "text/bbcode"];
+
+		if (!empty($item['signed_text']) && ($item['uri'] != $item['thr-parent'])) {
+			$data['diaspora:comment'] = $item['signed_text'];
+		}
+
+		$data['attachment'] = []; // @ToDo
+		$data['tag'] = self::createTagList($item);
+		$data = array_merge($data, self::createPermissionBlockForItem($item));
+
+		return $data;
+	}
+
+	/**
+	 * @brief Transmits a profile deletion to a given inbox
+	 *
+	 * @param integer $uid User ID
+	 * @param string $inbox Target inbox
+	 */
+	public static function transmitProfileDeletion($uid, $inbox)
+	{
+		$owner = User::getOwnerDataById($uid);
+		$profile = APContact::getByURL($owner['url']);
+
+		$data = ['@context' => 'https://www.w3.org/ns/activitystreams',
+			'id' => System::baseUrl() . '/activity/' . System::createGUID(),
+			'type' => 'Delete',
+			'actor' => $owner['url'],
+			'object' => $owner['url'],
+			'published' => DateTimeFormat::utcNow(DateTimeFormat::ATOM),
+			'to' => [ActivityPub::PUBLIC_COLLECTION],
+			'cc' => []];
+
+		$signed = LDSignature::sign($data, $owner);
+
+		logger('Deliver profile deletion for user ' . $uid . ' to ' . $inbox .' via ActivityPub', LOGGER_DEBUG);
+		HTTPSignature::transmit($signed, $inbox, $uid);
+	}
+
+	/**
+	 * @brief Transmits a profile change to a given inbox
+	 *
+	 * @param integer $uid User ID
+	 * @param string $inbox Target inbox
+	 */
+	public static function transmitProfileUpdate($uid, $inbox)
```

Comment:
Please rename to `sendProfileUpdate` to avoid repeating "transmit" when calling the method.